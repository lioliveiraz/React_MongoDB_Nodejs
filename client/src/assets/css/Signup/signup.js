import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
  root: {
    height: "100vh",
    background: "#BDBFBF",
  },
  main: {
    height: "70%",
    width: "80%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    background: "white",
    margin: "auto",
  },
  title: {
    padding: "1rem",
    fontWeight: "700",
  },
  inputs: {
    padding: "0.3rem",
  },
}));
