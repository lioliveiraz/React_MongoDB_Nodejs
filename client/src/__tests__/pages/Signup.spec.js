import React from "react";
import { mount, shallow, spyOn } from "enzyme";

import Signup from "./../../pages/Signup";
import { validateTruthiness } from "./../utils/testUtils";

describe("SignUp", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Signup />);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render correctly", () => {
    const signUpForm = wrapper.find("[data-cy='signUp-form']");
    const inputEmail = wrapper.find("[data-cy='input-email']");
    const inputPassword = wrapper.find("[data-cy='input-password']");
    const inputName = wrapper.find("[data-cy='input-name']");
    const button = wrapper.find("[data-cy='sign-up-submit-button']");
    const checkbox = wrapper.find("[data-cy='checkbox']");
    const title = wrapper.find("h1");

    const elementsArr = [
      signUpForm,
      inputEmail,
      inputPassword,
      inputName,
      checkbox,
      button,
      title,
    ];
    elementsArr.forEach((el) => {
      validateTruthiness(el);
    });
  });

  it("button should be disabled", () => {
    const button = wrapper.find("button").at(0);

    expect(button.prop("disabled")).toBeTruthy();
  });
  it("Should  password error message display", async () => {
    const inputEmail = wrapper.find("[data-cy='input-email']");
    const inputPassword = wrapper.find("[data-cy='input-password']");
    const inputName = wrapper.find("[data-cy='input-name']");
    const button = wrapper.find("button").at(0);

    await inputEmail.simulate("input", {
      target: { value: "email", name: "email" },
    });

    await inputPassword.simulate("input", {
      target: { value: "12345678", name: "name" },
    });

    await inputName.simulate("input", {
      target: { value: "joao", name: "name" },
    });
    await button.simulate("click");
    expect(wrapper.text()).toContain("must be a valid email");
  });
});
