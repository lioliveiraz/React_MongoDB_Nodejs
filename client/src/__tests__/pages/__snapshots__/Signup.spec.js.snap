// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `ReactWrapper {}`;

exports[`SignUp should match snapshot 2`] = `"<h1>Sign Up</h1><form data-cy=\\"signUp-form\\"><div class=\\"MuiFormControl-root MuiTextField-root makeStyles-root-1\\" data-cy=\\"input-name-wrapper\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-secondary\\" id=\\"standard-secondary-label\\">Name<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"standard-secondary\\" name=\\"name\\" placeholder=\\"John Doe\\" required=\\"\\" type=\\"text\\" data-cy=\\"input-name\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-4\\"><span>Name&nbsp;*</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root makeStyles-root-1\\" data-cy=\\"input-email-wrapper\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-secondary\\" id=\\"standard-secondary-label\\">Email<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"standard-secondary\\" name=\\"email\\" placeholder=\\"johndoe@email.com\\" required=\\"\\" type=\\"email\\" data-cy=\\"input-email\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-4\\"><span>Email&nbsp;*</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root makeStyles-root-1\\" data-cy=\\"input-password-wrapper\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-secondary\\" id=\\"standard-secondary-label\\">Password<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"standard-secondary\\" name=\\"password\\" required=\\"\\" type=\\"password\\" data-cy=\\"input-password\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-4\\"><span>Password&nbsp;*</span></legend></fieldset></div></div><input type=\\"checkbox\\" name=\\"adm\\" data-cy=\\"checkbox\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" data-cy=\\"sign-up-submit-button\\"><span class=\\"MuiButton-label\\"> Submit </span></button></form>"`;
